File: ui/lib/pyigor/domain/job.py
from pyigor.domain import utils


class Job:

    def __init__(self, data: dict):
        self.name = self.key = data.get("name")
        self.id = data.get("id")
        self.status = data.get("status")
        self.etag = data.get("etag")
        self.created_at = utils.human_time(data.get("created_at"))
        self.updated_at = utils.human_time(data.get("updated_at"))

File: pkg/structs/job.go
package structs

type JobSpec struct {
	// Optional
	Name string `json:"name"`
}

type Job struct {
	JobSpec `json:",inline"`

	ID     string `json:"id"`
	Status Status `json:"status"`
	ETag   string `json:"etag"`

	CreatedAt int64 `json:"created_at"`
	UpdatedAt int64 `json:"updated_at"`
}
